base/ca/src/com/netscape/ca/CAService.java

```diff
@@ -1374,6 +1383,43 @@ boolean verifySCT(CTResponse response, byte[] tbsCert, String logPublicKey) {
         }
     }
 
+    /**
+     * parses and gleans the hashing algorithm from the returned
+     * CT signature
+     * cfu
+     */
+    /*
+       enum HashAlgorithm {
+            none, md5, sha1, sha224, sha256, sha384, sha512};
+    */
+    enum HashAlgorithm {none, MD5, SHA1, SHA224, SHA256, SHA384, SHA512};
+    public String getHashAlgFromSig(byte[] ct_signature) {
+
+        int hashingAlg = Byte.toUnsignedInt(ct_signature[0]);
+        if (hashingAlg < 4 || hashingAlg > 5)
```

Comment:
Currently, `SHA 256` and `SHA384` seems to be valid. But, according to RCS only SHA256 is used in v1.

Can this condition be simplified to `if (hashingAlg != 4) return null` ?

Reply:
sure. I could do that.